host-controller/src/main/java/org/jboss/as/domain/controller/operations/SyncServerStateOperationHandler.java

```diff
@@ -126,30 +148,51 @@ public void handleResult(OperationContext.ResultAction resultAction, OperationCo
         }, OperationContext.Stage.MODEL);
     }
 
-    private Map<String, Boolean> determineServerStateChanges(OperationContext context, Resource domainRootResource, ServerOperationResolver resolver,
-                                                             ContentDownloader contentDownloader) {
-        final Map<String, Boolean> serverStateChanges = new HashMap<>();
+    private Map<String, SyncServerResultAction> determineServerStateChanges(final OperationContext context, final Resource domainRootResource, final ServerOperationResolver resolver,
+                                                                            final ContentDownloader contentDownloader) {
+        final Map<String, SyncServerResultAction> serverStateChanges = new HashMap<>();
         for (ModelNode operation : operations) {
-            PathAddress addr = PathAddress.pathAddress(operation.get(OP_ADDR));
+            final PathAddress addr = PathAddress.pathAddress(operation.get(OP_ADDR));
             contentDownloader.checkContent(operation, addr);
-
-            Map<Set<ServerIdentity>, ModelNode> serverMap = resolver.getServerOperations(context, operation, addr);
+            final ImmutableManagementResourceRegistration registration = context.getResourceRegistration();
+            final Map<Set<ServerIdentity>, ModelNode> serverMap = resolver.getServerOperations(context, operation, addr);
             for (Map.Entry<Set<ServerIdentity>, ModelNode> entry : serverMap.entrySet()) {
-                ModelNode op = entry.getValue();
-                boolean restart = op.get(OP).asString().equals(ServerProcessStateHandler.REQUIRE_RESTART_OPERATION);
```

Comment:
Ah, this makes a lot more sense now. I couldn't ever see the ServerProcessStateHandler.REQUIRE_RESTART_OPERATION arriving, so I had suspected that it was just wrong.

That makes a lot more sense now!